compiler/rustc_smir/src/stable_mir/mir/body.rs

```diff
@@ -133,9 +133,90 @@ pub enum AsyncGeneratorKind {
     Fn,
 }
 
+// FIXME: We are using `usize` for now but we should be using DefIds and find what
+//        what those are refering to and then use appropirate ty_defs for them (i.e AdtDef)
```

Comment:
Please just use the `opaque` function instead of encoding information that is useless to SMIR users